src/compiler/abstract_compiler.nit

```diff
@@ -2689,10 +2709,20 @@ redef class AMethPropdef
 		var pname = mpropdef.mproperty.name
 		var cname = mpropdef.mclassdef.mclass.name
 		var ret = mpropdef.msignature.return_mtype
-		if ret != null then
+		var compiler = v.compiler
+
+		# WARNING: we must not resolve the return type when it's a functional type.
+		# Otherwise, we get a compile error exactly here. This weird behavior doesn't affect
+		# the inner mecanics of callref since the return type is already solved by
+		# `routine_ref_call`
+		if ret != null and not compiler.all_routine_types_name.has(cname) then
```

Comment:
This is really weird. Do you have a test-case? I cannot reproduce a case where the additional condition is needed.